internal/nix/source.go

```diff
@@ -0,0 +1,52 @@
+package nix
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func SourceNixEnv() error {
+	// if command is not in path, the source the nix startup files and hopefully
+	// the command will be found. (we should still check that nix is actually
+	// installed before we get here)
+	srcFile := "/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh"
+	// if global (multi-user) daemon file is missing, try getting the single user
+	// file.
+	if _, err := os.Stat(srcFile); os.IsNotExist(err) {
+		srcFile = filepath.Join(
+			os.Getenv("HOME"),
+			"/.nix-profile/etc/profile.d/nix.sh",
+		)
+	}
+
+	cmd := exec.Command(
+		"bash",
+		"-c",
+		fmt.Sprintf("source %s ; echo '<<<ENVIRONMENT>>>' ; env", srcFile),
```

Comment:
Will this work with `/bin/sh`? bash isn't available in Alpine images, for example. 

I'd also suggest clearing the environment to try and grab just the environment variables exported by the script (maybe even do a before-and-after diff). Otherwise we may end up reverting env vars that the user changed since starting their terminal.

```suggestion
		"env",
		"-i",
		"/bin/sh",
		"-c",
		fmt.Sprintf(`. "%s" ; echo '<<<ENVIRONMENT>>>' ; env`, srcFile),
```

Reply:
Good calls.

Reply:
@gcurtis one issue your solution runs into is that the nix script itself uses `__ETC_PROFILE_NIX_SOURCED` to determine if it should run again. I can always pass that variable manually, but that's a bit fragile.

Question about variables changing, the commands Env should be exactly the same as the current process, from the docs:

```
// If Env is nil, the new process uses the current process's
// environment.
```

so the diff should in theory be just the loaded ones right?

Reply:
I'm switching bash to sh, but I don't think we can use a fresh environment. The reason is that the sourced files use environment variables (PATH and __ETC_PROFILE_NIX_SOURCED) this means that we would get bad values.

I actually think the risk of a stale variable is minimal because how this is used. It only gets called at pre-command time if `/nix` directory is found but nix is not in path. It never gets called again.

Reply:
Yeah, good point. Alright, all sounds good to me!